apiserver/facade/interface.go

```diff
@@ -139,13 +141,13 @@ type Context interface {
 // ControllerDBGetter defines an interface for getting the controller DB.
 type ControllerDBGetter interface {
 	// ControllerDB returns a transaction runner for the controller database.
-	ControllerDB() (coredatabase.TxnRunner, error)
+	ControllerDB() (changestream.WatchableDB, error)
 }
 
-// DBManager defines an interface for managing the database.
+// DBManager describes the ability to acquire a database deleter.
 type DBManager interface {
-	// DBManager returns a DBManager reference for the controller database.
-	DBManager() coredatabase.DBManager
+	// DBDeleter returns an a deleter for databases based on namespace.
+	DBDeleter() coredatabase.DBDeleter
```

Comment:
do we still need this `DBManager` interface? couldn't we just use `database.DBDeleter`? feels like too much indirection

Reply:
DBManager is gone.

Reply:
I'm talking about the DBManager interface that contains the `DBDeleter()` method, is it necessary?:

```
// DBManager describes the ability to acquire a database deleter.
type DBManager interface {
	// DBDeleter returns an a deleter for databases based on namespace.
	DBDeleter() coredatabase.DBDeleter
}
```

Reply:
Ah, I see. I'm happy to keep this for now. 

I embedded it in the facade to make the registration code simpler for `modelmanager`. I _assume_ it was as it was (casting the context to the interface) so that it is not trivially available to any and all. It may need to change again, but I'll do that in a subsequent patch if so.